File: tests/fixtures/cpp_sample/src/circle.cpp
#include "circle.h"
#include <vector>

Circle::Circle(double r) : radius(r) {}

double Circle::area() const {
    return 3.14159 * radius * radius;
}

namespace {
double anon_helper(double x) { return x * x; }
}

using StringList = std::vector<int>;

File: tests/fixtures/rust_sample/src/geometry/circle.rs
use super::Shape;

pub struct Circle {
    pub radius: f64,
}

impl Circle {
    pub fn new(radius: f64) -> Self {
        Circle { radius }
    }
}

impl Shape for Circle {
    fn area(&self) -> f64 {
        3.14159 * self.radius * self.radius
    }
}

pub fn unit_circle() -> Circle {
    Circle::new(1.0)
}
